client/document/index.jsx

```diff
@@ -45,8 +46,28 @@ class Document extends React.Component {
 			devDocs,
 			devDocsURL,
 			feedbackURL,
+			inlineScriptNonce,
+			analyticsScriptNonce,
 		} = this.props;
 
+		let inlineScript = `COMMIT_SHA = ${ jsonStringifyForHtml( commitSha ) };\n`;
+
+		if ( user ) {
+			inlineScript += `var currentUser = ${ jsonStringifyForHtml( user ) };\n`;
+		}
+
+		if ( app ) {
+			inlineScript += `var app = ${ jsonStringifyForHtml( app ) };\n`;
+		}
+
+		if ( initialReduxState ) {
+			inlineScript += `var initialReduxState = ${ jsonStringifyForHtml( initialReduxState ) };\n`;
+		}
+
+		if ( config ) {
+			inlineScript += config;
+		}
```

Comment:
what's the purpose of these changes? they don't seem to be related to the CSP behaviors but maybe there's something obvious that isn't jumping out…

Reply:
oh now I see it's to add the nonce, I think

how would you feel about also eliminating the mutation on the string. it's not necessary in any case

```js
const inlineScript = (
	`COMMIT_SHA = ${ … };\n` +
	( user ? `var currentUser = ${ … };\n` : '' ) +
	( app ? … : … ) +
	…
)
```

The thing I like about that is not having `inlineScript` ever in an invalid state and also not having to trace through the code to see if we might end up changing it lower down unexpectedly. As a `const` string we know it won't.

Reply:
> oh now I see it's to add the nonce, I think

Indeed

> how would you feel about also eliminating the mutation on the string. it's not necessary in any case

Great idea, I'll update the PR